driver-core/src/main/com/mongodb/MongoWriteConcernException.java

```diff
@@ -52,24 +57,31 @@ public MongoWriteConcernException(final WriteConcernError writeConcernError, fin
      * @param writeConcernResult the write result
      * @param serverAddress     the non-null server address
      * @since 3.2
+     * @deprecated Prefer {@link MongoWriteConcernException(WriteConcernError, WriteConcernResult, ServerAddress, Set)}
      */
-    @SuppressWarnings("deprecation")
+    @Deprecated
     public MongoWriteConcernException(final WriteConcernError writeConcernError, @Nullable final WriteConcernResult writeConcernResult,
                                       final ServerAddress serverAddress) {
+        this(writeConcernError, writeConcernResult, serverAddress, Collections.emptySet());
+    }
+
+    /**
+     * Construct an instance.
+     *
+     * @param writeConcernError the non-null write concern error
+     * @param writeConcernResult the write result
+     * @param serverAddress     the non-null server address
+     * @param errorLabels       the server errorLabels
+     * @since 5.0
+     */
+    public MongoWriteConcernException(final WriteConcernError writeConcernError, @Nullable final WriteConcernResult writeConcernResult,
+            final ServerAddress serverAddress, final Set<String> errorLabels) {
```

Comment:
Note how `MongoException.addLabels` accepts `Collection`, not `Set`: all the method needs is the ability to iterate the given `errorLabels` (arguably, it should have been `Iterable`). Thus, I don't think there is a reason for us to restrict the type of `errorLabels` to `Set`, we can also accept `Collection` here.

This comment is also applicable to the new constructor of `MongoWriteException`.

Reply:
`MongoException.addLabels` probably should have been accepted a Set, since ultimately duplicates are discarded.  Since it's protected it's not a big deal to change its signature.  What do you think?

Reply:
The absence of duplicates is important only when labels are read, and is expressed by `getErrorLabels` returning `Set`. By requiring a `Set` in `addLabels` (or in the new constructor), we are coercing a user to convert from whatever container he has into a `Set`, seemingly for no good reason: we are still relying only on being able to iterate through the elements of the argument of the `addLabels` method.

It's not an important thing, so I marked the thread as optional in case you are not convinced.

Reply:
I buy the argument. Done.